rootex/framework/components/visual/camera_component.cpp

```diff
@@ -73,6 +80,10 @@ void from_json(const JSON::json& j, PostProcessingDetails& p)
 	p.toneMapOperator = j.at("toneMapOperator");
 	p.toneMapTransferFunction = j.at("toneMapTransferFunction");
 	p.toneMapWhiteNits = j.at("toneMapWhiteNits");
+	for (auto&& customPostProcessing : j.at("customPostProcessing").items())
+	{
+		p.m_CustomPostProcessing.insert({ (String)customPostProcessing.key(), (bool)customPostProcessing.value() });
```

Comment:
When a camera component is loaded, custom post processing details are being inserted to `m_CustomPostProcessing`. But the only call to `addCustomPostProcessing` is from ImGUI. Hence when a level is loaded objects for these saved custom post processing effects aren't created.

Call the function wherever `PostProcessingDetails& p` is being used or here itself would be fine as well.

also I think `PostProcessingDetails` is a struct so no need for the `m_` convention.

Reply:
fixed